subprojects/core/src/main/java/org/gradle/api/internal/tasks/SnapshotTaskInputsOperationDetails.java

```diff
@@ -28,20 +30,19 @@
  *
  * This operation is executed only when the build cache is enabled.
  *
- * This class is intentionally internal and consumed by the build scan plugin.
- *
  * @since 4.0
  */
+@UsedByScanPlugin
 public final class SnapshotTaskInputsOperationDetails implements BuildOperationDetails<SnapshotTaskInputsOperationDetails.Result> {
 
-    private final String taskPath;
+    private final Task task;
 
-    public SnapshotTaskInputsOperationDetails(String taskPath) {
-        this.taskPath = taskPath;
+    public SnapshotTaskInputsOperationDetails(Task task) {
+        this.task = task;
     }
 
-    public String getTaskPath() {
-        return taskPath;
+    public Task getTask() {
```

Comment:
We shouldn't leak mutable state into operation details. Can you instead expose the details you need from the task?